src/components/NotificationsDrawer/DrawerPanelContent.tsx

```diff
@@ -39,19 +57,150 @@ const EmptyNotifications = () => (
+    const modifiedNotifications = (activeFilters || []).reduce(
+      (acc: NotificationData[], chosenFilter: string) => [...acc, ...notifications.filter(({ source }) => source === chosenFilter)],
+      []
+    );
+
+    setFilteredNotifications(modifiedNotifications);
+  }, [activeFilters]);
+
+  const onNotificationsDrawerClose = () => {
+    setActiveFilters([]);
+    dispatch(toggleNotificationsDrawer());
+  };
+
+  const onMarkAllAsRead = () => {
+    dispatch(markAllNotificationsAsRead());
+    setIsDropdownOpen(false);
+  };
+
+  const onMarkAllAsUnread = () => {
+    dispatch(markAllNotificationsAsUnread());
+    setIsDropdownOpen(false);
+  };
+
+  const onFilterSelect = (chosenFilter: string) => {
+    activeFilters.includes(chosenFilter)
+      ? setActiveFilters(activeFilters.filter((filter) => filter !== chosenFilter))
+      : setActiveFilters([...activeFilters, chosenFilter]);
+  };
+
+  const dropdownItems = [
+    <DropdownItem key="read all" onClick={onMarkAllAsRead}>
+      Mark visible as read
+    </DropdownItem>,
+    <DropdownItem key="unread all" onClick={onMarkAllAsUnread}>
+      Mark visible as unread
+    </DropdownItem>,
+    <Divider />,
+    <DropdownItem key="event log">
+      <Icon>
+        <ExternalLinkSquareAltIcon />
+      </Icon>
+      View event log
+    </DropdownItem>,
+    <DropdownItem key="notification settings">
+      <Icon>
+        <ExternalLinkSquareAltIcon />
+      </Icon>
+      Configure notification settings
+    </DropdownItem>,
+    <DropdownItem key="notification preferences">
+      <Icon>
+        <ExternalLinkSquareAltIcon />
+      </Icon>
+      Manage my notification preferences
+    </DropdownItem>,
+  ];
+
+  const filterDropdownItems = () => {
+    const sources = notifications.reduce((acc: string[], { source }) => (acc.includes(source) ? acc : [...acc, source]), []);
+
+    return [
+      <DropdownGroup key="filter-label" label="Show notifications for...">
+        {sources.map((source, index) => (
+          <DropdownItem key={index} onClick={() => onFilterSelect(source)}>
+            <Checkbox isChecked={activeFilters.includes(source)} id={index.toString()} className="pf-u-mr-xs" />
+            {source}
+          </DropdownItem>
+        ))}
+        <Divider />
+        <DropdownItem key="reset-filters" onClick={() => setActiveFilters([])}>
+          <Button variant="link" isInline>
+            Reset filters
+          </Button>
+        </DropdownItem>
+      </DropdownGroup>,
+    ];
+  };
+
+  const renderNotifications = () => {
+    if (notifications.length === 0) {
+      return <EmptyNotifications />;
+    }
+
+    // TODO: Add sorting by timestamps as primary sort, then by read/unread.
+    const sortedNotifications = (filteredNotifications?.length > 0 ? filteredNotifications : notifications).sort(
+      (currentNotification, nextNotification) => (currentNotification.read === nextNotification.read ? 0 : currentNotification.read ? 1 : -1)
+    );
```

Comment:
We want to sort over two values, it's better to use `lodash`'s [sortBy](https://lodash.com/docs/4.17.15#sortBy) for this one